compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -459,6 +459,40 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                         )
                         .into()
                     }
+                    (&GenericParamDefKind::Const { has_default }, hir::GenericArg::Infer(inf)) => {
+                        if has_default {
+                            // FIXME(const_generics): Actually infer parameter here?
+                            tcx.const_param_default(param.def_id).into()
+                        } else {
```

Comment:
need a `allow_ty_infer` check here. Probably renaming `allow_ty_infer` to something which encompasses both types and consts

Reply:
I've left this comment up as I'm not quite sure what to do with the default, as we don't allow const inference yet, I believe? I just left a `todo!()` in the code temporarily

Reply:
this is still an open question